gcloud-java-examples/src/main/java/com/google/cloud/examples/logging/snippets/LoggingSnippets.java

```diff
@@ -0,0 +1,545 @@
+  // [VARIABLE "my_metric_name"]
+  public boolean deleteMetric(String metricName) {
+    // [START deleteMetric]
+    boolean deleted = logging.deleteMetric(metricName);
+    if (deleted) {
+      // the metric was deleted
+    } else {
+      // the metric was not found
+    }
+    // [END deleteMetric]
+    return deleted;
+  }
+
+  /**
+   * Example of asynchronously deleting a metric.
+   */
+  // [TARGET deleteMetricAsync(String)]
+  // [VARIABLE "my_metric_name"]
+  public boolean deleteMetricAsync(String metricName)
+      throws ExecutionException, InterruptedException {
+    // [START deleteMetricAsync]
+    Future<Boolean> future = logging.deleteMetricAsync(metricName);
+    // ...
+    boolean deleted = future.get();
+    if (deleted) {
+      // the metric was deleted
+    } else {
+      // the metric was not found
+    }
+    // [END deleteMetricAsync]
+    return deleted;
+  }
+
+  /**
+   * Example of writing log entries and providing a default log name and monitored resource.
+   */
+  // [TARGET write(Iterable, WriteOption...)]
+  // [VARIABLE "my_log_name"]
+  public void write(String logName) {
+    // [START write]
+    List<LogEntry> entries = new ArrayList<>();
+    entries.add(LogEntry.of(StringPayload.of("Entry payload")));
+    Map<String, Object> jsonMap = new HashMap<>();
+    jsonMap.put("key", "value");
+    entries.add(LogEntry.of(JsonPayload.of(jsonMap)));
+    logging.write(entries,
+        WriteOption.logName(logName),
+        WriteOption.resource(MonitoredResource.builder("global").build()));
+    // [END write]
+  }
+
+  /**
+   * Example of asynchronously writing log entries and providing a default log name and monitored
+   * resource.
+   */
+  // [TARGET writeAsync(Iterable, WriteOption...)]
+  // [VARIABLE "my_log_name"]
+  public Future<Void> writeAsync(String logName) {
+    // [START writeAsync]
+    List<LogEntry> entries = new ArrayList<>();
+    entries.add(LogEntry.of(StringPayload.of("Entry payload")));
+    Map<String, Object> jsonMap = new HashMap<>();
+    jsonMap.put("key", "value");
+    entries.add(LogEntry.of(JsonPayload.of(jsonMap)));
+    Future<Void> future = logging.writeAsync(
+        entries,
+        WriteOption.logName(logName),
+        WriteOption.resource(MonitoredResource.builder("global").build()));
+    // [END writeAsync]
+    return future;
+  }
+
+  /**
+   * Example of listing log entries for a specific log.
+   */
+  // [TARGET listLogEntries(EntryListOption...)]
+  // [VARIABLE "logName=projects/my_project_id/logs/my_log_name"]
+  public Page<LogEntry> listLogEntries(String filter) {
+    // [START listLogEntries]
+    Page<LogEntry> entries = logging.listLogEntries(EntryListOption.filter(filter));
+    Iterator<LogEntry> entryIterator = entries.iterateAll();
+    while (entryIterator.hasNext()) {
+      LogEntry entry = entryIterator.next();
+      // do something with the entry
+    }
+    // [END listLogEntries]
+    return entries;
```

Comment:
Is this still useful at the end?  Can I create another Iterator and get everything?